src/applications/widget-editor/src/sagas/widget/index.js

```diff
@@ -95,24 +101,26 @@ function* preloadData() {
   }
 }
 
-function* resolveWithProxy(payload) {
+// Called multiple, checks if we need to update data
+// Using @core/stateProxy
+// If no updates required we yield cancel
+function* checkWithProxyIfShouldUpdate(payload) {
   const { widgetEditor } = yield select();
 
   // We only want to resolve proxy if the editor itself is initialized
   if (!widgetEditor.editor.initialized) {
     yield cancel();
-  } 
-  
-  // Check and patch current state based on user configuration
+  }
+
+  // State proxy checks editor state if any data updates are required.
+  // state proxy returns array of actions that we call to update the editor
   const proxyResult = yield call([stateProxy, "sync"], widgetEditor, payload.type);
-  // --- Proxy results returns a list of events we call on certain updates
-  // --- This makes sure we only update what is nessesary in the editor
+
   if (proxyResult && proxyResult.length > 0) {
     for (const evnt in proxyResult) {
       yield put({ type: proxyResult[evnt] });
     }
-    const state = yield select();
-    localOnChangeState(state.widgetEditor);
+    yield call(updateHookState);
```

Comment:
no need to dispatch `DATA_FLOW_UPDATE_HOOK_STATE` as we can just yield the generator here.